### ISSUE TITLE
Optional cuda/dnn?

### ISSUE THREAD

**Issue by atanisoft** [zigcv/zigcv#52]
Is it possible to make cuda and dnn optional includes? These are not available by default on Gentoo and both depend solely on Nvidia cards locking the functionality out of AMD users.

Perhaps adding opencl as alternative?

**Comment by ryoppippi** [comment]
@atanisoft 
Sorry for my late reply?
So currently because of cuda import, you cannot use zigcv right?

**Comment by atanisoft** [comment]
Yup, it's not included by default on Gentoo sadly. I tried to included it but ran into a few other blockers on the Gentoo side and didn't go too much further on that aspect.

**Comment by ryoppippi** [comment]
I see!
Can you send a PR?

**Comment by atanisoft** [comment]
I'll need to dig into that, I've got a couple items on my plate taking priority but once I get those out of the way I'll see what I can come up with.

I'm thinking some build options that can be used to include/exclude features, these would be passed into the code to include/exclude imports. At least that is my initial thinking, the options can be on by default and have an option to disable I'm thinking.  Thoughts?
